Feat/product crud base: service & dao layer, mapper 구현

sanghyeonJwa opened the pull request.

sanghyeonJwa committed:
#12 상품 조회 & 필터링 기능 개발 완료

1. 상품 조회 기능

2. 상품 필터링 기능
[필터링 목록]
1. 상품 명
2. 카테고리
3. 가격 (최소, 최대)
4. 판매 여부
5. 날짜 기간

sanghyeonJwa committed:
#12 form 태그 디자인 수정

sanghyeonJwa committed:
#13 feat: 신상품 등록 Service & Dao, Mapper layer 구현

신상품 등록 Service & Dao, Mapper layer 구현
- 완료하기 위해 InventoryMapper 필요

sanghyeonJwa committed:
#14 #15 feat: 상품 정보 변경, 상품 정보 삭제(비활성화) DAO, service layer 구현

상품 정보 변경, 상품 정보 삭제(비활성화) DAO, service layer 구현

- 해당 커밋을 분리했어야 했는데 죄송합니다. 앞으로는 분리하겠습니다!

sanghyeonJwa committed:
#28 refactor: 도메인 클래스 삭제 및 image dto 패키지 구조, 명칭 변경

처음엔 도메인 클래스에 넣어서 db 접근하려고 했는데 문제점이 좀 있어서
변경했습니다..

sanghyeonJwa committed:
#17 feat: 카테고리 추가 service, dao layer & mapper.xml 구현

카테고리 추가 service, dao layer & mapper.xml 구현

sanghyeonJwa committed:
#29 feat: 카테고리 수정 service, dao layer & mapper.xml 구현

카테고리 수정 service, dao layer & mapper.xml 구현
- 삭제 부분은 db 데이터 변경 및 상품 로직 부분도 함께 되어 있어서 논의가 필요할 것 같습니다.

kebin0591 commented on src/main/java/com/nf404/devshop/product/service/ProductService.java:
서비스 단에서 예외처리 같은 것도 나중에 추가하면 좋을 것 같습니다! 고생하셨습니다!

sanghyeonJwa replied:
넵 알겠습니다!

DokaDev commented on src/main/resources/mappers/product/CategoryMapper.xml:
```diff
@@ -0,0 +1,40 @@
+        WHERE ref_category_code IS NULL
+    </select>
+
+    <select id="getByRefCategoryCode" resultMap="categoryResultMap">
```
id명만으로는 dao 메소드의 동작을 유추하기가 다소 어렵습니다. 이름 조정부탁드립니다.

DokaDev replied:
selectSubCategoriesByParentCategoryCode()

sanghyeonJwa replied:
좀더 직관적인 이름으로 수정했습니다!

DokaDev commented on src/main/resources/mappers/product/CategoryMapper.xml:
```diff
@@ -0,0 +1,40 @@
+        SELECT * FROM tbl_category
+    </select>
+
+    <insert id="insertCategory">
```
parameterType="com.....xxxDto"

xxx.ref_category_code;

sanghyeonJwa replied:
현재 이 부분은 참조하는 parameter가 1개일 경우 자동으로 mapping 되는 것으로 알고 있어서 추후에 문제가 생기거나 코드 로직을 이해하는데 어렵다면 추가하겠습니다!

sanghyeonJwa committed:
refactor(#28): 상품 도메인 패키지 구조 변경, 엔티티 클래스 추가, dto 필드값 변경

1. 상품 도메인에서 dto와 entity 클래스 model 패키지 안에 담기도록 패키지 구조 변경
2. 엔티티 클래스 추가
3. 기존에 dto에 다른 테이블의 필드 값을 넣었던 것을 객체로 변경 (Category, Image)

sanghyeonJwa committed:
refactor(#12 #13 #14): dao, service, mapper.xml 로직 변경

1. Mapper의 명칭을 보다 더 직관적이게 변경
2. Mapper.xml 반환 resultMap 변경 및 로직 변경
- ImageMapper.xml 의 updateImageInfo 에 join 로직 대체
- ProductMapper.xml 에 상품 코드로 상품 데이터 반환 받는 mapper 추가

kebin0591 reviewed (approved):
수정하신 내용 확인했습니다!

sanghyeonJwa committed:
feat(#12, #13): 상품 조회 시 이미지 불러오기 기능 추가, 신상품 등록 구현 완료, 이미지 처리 로직 구현 완료

- 이미지 저장 밑 이름 변경 로직 global의 utility 패키지 안에 넣어놨습니다.
- 신상품 등록 service 단에 재고 테이블도 만드는 로직 추가가 필요합니다.

sanghyeonJwa committed:
refactor(#28): CategoryController import 부분 경로 변경
